jujugui/static/gui/src/app/components/user-profile/entity/entity.js

```diff
@@ -281,6 +292,77 @@ YUI.add('user-profile-entity', function() {
       );
     },
 
+    _getMetrics: function(filters) {
+      if (this.state.hasMetrics) {
+        return;
+      }
+      // TODO filters such as summary-interval and start/end dates
+      // can be passed in here, but will need UX first.
+      // Makyo - 2017-03-10
+      this.props.getKpiMetrics(
+          // XXX Metrics are not fully implemented in Omnibus yet. Until they
+          // are, a dummy charm is used. The correct implementation is
+          // commented out below, with the dummy charm in its place.
+          // To QA, uncomment the following line and comment out the line
+          // after that.
+          // Makyo - 2017-03-17
+          //'cs:~canonical/jimm-0',
+          this.props.entity.id,
+          filters,
+          (error, data) => {
+            if (!error && data.length > 0) {
+              let metrics = [];
+              let metricTypes = this.state.metricTypes;
+              data.forEach((item) => {
+                if (metricTypes.indexOf(item.Metric) === -1) {
+                  metricTypes.push(item.Metric);
+                }
+                metrics.push(item);
+              });
+              this.setState({
+                hasMetrics: true,
+                metrics: metrics,
+                metricTypes: metricTypes
+              });
+            } else {
+              this.setState({
+                hasMetrics: false,
+                metrics: {},
+                metricTypes: []
+              });
+            }
+          });
+    },
+
+    _showMetrics: function() {
+      return (
+        <juju.components.UserProfileEntityKPI
+          charmId={this.props.entity.id}
+          d3={this.props.d3}
+          metrics={this.state.metrics}
+          metricTypes={this.state.metricTypes} />);
+    },
+
+    _toggleKpiVisibility: function(e) {
+      this.setState({
+        kpiVisible: !this.state.kpiVisible
+      });
+    },
+
+    _generateMetrics: function() {
+      if (this.props.type === 'charm') {
+        return (
+          <div>
+            <juju.components.GenericButton
+              title={this.state.kpiMetrics ? 
+                'Hide KPI Metrics' : 'Show KPI Metrics'}
+              action={this._toggleKpiVisibility} />
+            {this.state.kpiVisible ? this._showMetrics() : undefined}
```

Comment:
This is counter to what you told me while pairing. You told me to gate an expensive method behind a state var through a switch in the JSX. Holding off for further clarification.